MobileBuy/buy/src/main/java/com/shopify/buy/utils/AndroidPayHelper.java

```diff
@@ -0,0 +1,391 @@
+        if (checkout == null) {
+            throw new NullPointerException("checkout cannot be null");
+        }
+
+        if (maskedWallet == null) {
+            throw new NullPointerException("maskedWallet cannot be null");
+        }
+
+        return FullWalletRequest.newBuilder()
+                .setGoogleTransactionId(maskedWallet.getGoogleTransactionId())
+                .setCart(createWalletCart(checkout))
+                .build();
+    }
+
+    /**
+     * Will update the local checkout with the Addresses and email from a Masked Wallet.
+     * If any of the City, Country Code, Province Code, or ZIP code have changed any shipping rate previously set
+     * on the Checkout will be invalid and set to null. The checkout should be updated on Shopify using {@link BuyClient#updateCheckout(Checkout, Callback)},
+     * and the shipping rate will need to be fetched again using {@link BuyClient#getShippingRates(String, Callback)}
+     *
+     * @param checkout     The {@link Checkout} to update
+     * @param maskedWallet The {@link MaskedWallet} to use the address from
+     */
+    public static void updateCheckoutAddressAndEmail(Checkout checkout, MaskedWallet maskedWallet) {
+        if (checkout == null) {
+            throw new NullPointerException("checkout cannot be null");
+        }
+
+        if (maskedWallet == null) {
+            throw new NullPointerException("maskedWallet cannot be null");
+        }
+
+        Address shippingAddress = createShopifyAddress(maskedWallet.getBuyerShippingAddress());
+        Address billingAddress = createShopifyAddress(maskedWallet.getBuyerBillingAddress());
+
+        // If the location has changed, we need to invalidate the shipping address
+        if (!shippingAddress.locationsAreEqual(checkout.getShippingAddress())) {
+            checkout.setShippingRate(null);
+        }
+
+        checkout.setShippingAddress(shippingAddress);
+        checkout.setBillingAddress(billingAddress);
+        checkout.setEmail(maskedWallet.getEmail());
+    }
+
+    /**
+     * Checks to see if Android Pay is available on device.
+     *
+     * It will check that:
+     * 1) Play Services are available using {@link AndroidPayHelper#hasGooglePlayServices(Context)}
+     * 2) The BuyClient has Android Pay enabled using {@link BuyClient#androidPayIsEnabled()}
+     * 3) The Android Pay application is installed on device, and user has setup a valid card for In App Purchase using {@link AndroidPayHelper#isReadyToPay(GoogleApiClient, AndroidPayAvailableResponse)}
+     *
+     * @param context
+     * @param buyClient The {@link BuyClient} to use, not null
+     * @param apiClient The {@link GoogleApiClient}, not null
+     * @param delegate  The {@link AndroidPayAvailableResponse} delegate for receiving the result
+     */
+    public static void androidPayIsAvailable(final Context context, final BuyClient buyClient, final GoogleApiClient apiClient, final AndroidPayAvailableResponse delegate) {
```

Comment:
The same here, I would remove the buyClient param. This method should check only availability of android pay from OS perspective (check if google play service installed and ready to pay) but not if it is enabled / disabled inside buy client. To make helper be less depended from buy client.